src/background/sidebar-injector.ts

```diff
@@ -0,0 +1,485 @@
+import { chromeAPI, executeFunction, executeScript } from './chrome-api';
+import { detectContentType } from './detect-content-type';
+import {
+  AlreadyInjectedError,
+  BlockedSiteError,
+  LocalFileError,
+  NoFileAccessError,
+  RestrictedProtocolError,
+} from './errors';
+
+const CONTENT_TYPE_HTML = 'HTML';
+const CONTENT_TYPE_PDF = 'PDF';
+const CONTENT_TYPE_VITALSOURCE = 'VITALSOURCE';
+const CONTENT_TYPE_LMS = 'LMS';
```

Comment:
Maybe these could be "merged" into an `enum`.

Reply:
Yes. I think I'd like to leave that as a separate task since it involves changes across several modules.